### PR TITLE
benchmarks: switch to BenchmarkDotNet

### PR DESCRIPTION
**Pull request by omsmith**

### CODE REVIEW

**Comment by omsmith** [comment]
cherry-picked from `omsmith/keep`

File: test/D2L.Security.OAuth2.Benchmarks/D2L.Security.OAuth2.Benchmarks.csproj [hunk 1]
```diff
@@ -1,111 +1,34 @@
...
-    <ProjectGuid>{32C50F4A-03AE-4C99-9108-3AD518729410}</ProjectGuid>
-    <OutputType>Exe</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>D2L.Security.OAuth2.Benchmarks</RootNamespace>
-    <AssemblyName>D2L.Security.OAuth2.Benchmarks</AssemblyName>
-    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
-    <TargetFrameworkProfile />
+    <TargetFramework>net472</TargetFramework>
+    <LangVersion>7.3</LangVersion>
```
Comment (j3parker) [inline test/D2L.Security.OAuth2.Benchmarks/D2L.Security.OAuth2.Benchmarks.csproj:5]: Not going to go wild? ;)

File: test/D2L.Security.OAuth2.Benchmarks/FullStackValidation/FullStackValidationBenchmarks.cs [hunk 2]
```diff
@@ -1,27 +1,49 @@
 ﻿using System;
-using BenchmarkIt;
+using System.Collections.Generic;
+using BenchmarkDotNet.Attributes;
```
Comment (j3parker) [inline test/D2L.Security.OAuth2.Benchmarks/FullStackValidation/FullStackValidationBenchmarks.cs:3]: For .NET Standard?

Comment (omsmith) [reply test/D2L.Security.OAuth2.Benchmarks/FullStackValidation/FullStackValidationBenchmarks.cs:3]: Like why switch? Yeah I think so?

The commit is almost 2 years old. https://github.com/Brightspace/D2L.Security.OAuth2/commit/9de36e5db5d352ec46bfe4ede476f8ec3d42b7c1

Comment (j3parker) [reply test/D2L.Security.OAuth2.Benchmarks/FullStackValidation/FullStackValidationBenchmarks.cs:3]: Haha

**Review by j3parker — APPROVED** [review]

**Pull request merged**